### DISCUSSION TITLE
How to simplify the logic expression including xor('^')

### DISCUSSION THREAD

**Discussion by Flians** [sympy/sympy#25874] [Q&A]
Hello, could you please tell me how to simplify the logic expression including xor('^'), such as "((a | b) & c) | (a ^ c)"?
Now I carry out the command "simplify", and get the error "TypeError: expecting bool or Boolean, not `a**c`".

**Comment by asmeurer** [comment]
You should create the expressions directly. If you use a string, `^` is parsed as exponentiation. See https://docs.sympy.org/dev/explanation/best-practices.html#avoid-string-inputs

From there, you should use `simplify_logic`.
